django/contrib/admindocs/utils.py

```diff
@@ -239,3 +242,57 @@ def remove_non_capturing_groups(pattern):
         final_pattern += pattern[prev_end:start]
         prev_end = end
     return final_pattern + pattern[prev_end:]
+
+
+# `ThreadCallbackStrs` object holds instances of the _callback_strs as c_strs
+# objects that are local to the thread.
+class ThreadCallbackStrs(threading.local):
+    c_strs = set()
```

Comment:
`c_strs` isn't a clear name - please use something like `callback_strs`. (Closer to the original.)

Reply:
Also, I don't think this is entirely correct - previously the values were stored on the `URLResolver` itself. There may be different resolvers, e.g. if `request.urlconf` is set in middleware. We need to ensure that multiple can be handled.